tests/Latte/CoreMacros.templateType.php74.phpt

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/**
+ * Test: Latte\CoreMacros: {templateType ClassWithTypes}
+ */
+
+declare(strict_types=1);
+
+use Latte\Macros\CoreMacros;
+use Tester\Assert;
+use Tester\Environment;
+use Tester\FileMock;
+
+require __DIR__ . '/../bootstrap.php';
+
+if (PHP_VERSION_ID < 70400) {
+	Environment::skip('Typed properties cannot be tested with PHP < 7.4');
+}
+
+$compiler = new Latte\Compiler;
+CoreMacros::install($compiler);
+
+test(function () use ($compiler) {
+	require FileMock::create('
+	class TemplateTypeClass
```

Comment:
Not sure if it's possible to define class conditionally. Test fails for php < 7.4 before Environment::skip() is called